mobilecoind/src/payments.rs

```diff
@@ -131,29 +131,20 @@ impl<T: BlockchainConnection + UserTxConnection + 'static, FPR: FogPubkeyResolve
     }
 }
 
-fn get_network_block_version(block_infos: &[BlockInfo]) -> u32 {
-    block_infos
-        .iter()
-        .map(|block_info| block_info.network_block_version)
-        .max()
-        .unwrap_or(0)
-}
-
-fn get_fee(block_infos: &[BlockInfo], token_id: TokenId, opt_fee: u64) -> u64 {
-    if opt_fee > 0 {
-        opt_fee
-    } else if block_infos.is_empty() {
-        FALLBACK_FEE
-    } else {
-        // iterate an owned list of connections in parallel, get the block info for
-        // each, and extract the fee. If no fees are returned, use the hard-coded
-        // minimum.
-        block_infos
-            .iter()
-            .filter_map(|block_info| block_info.minimum_fee_or_none(&token_id))
-            .max()
-            .unwrap_or(FALLBACK_FEE)
+/// Get the most common BlockInfo out of a list of BlockInfos.
+/// The assumption is that in the majority of cases, all the BlockInfos
+/// would be the same. They will only differ during a network upgrade
+/// or if a node is left running with an old configuration.
+fn get_majority_block_info(block_infos: &[BlockInfo]) -> Option<BlockInfo> {
+    let mut block_info_counts = HashMap::default();
+    for block_info in block_infos {
+        *block_info_counts.entry(block_info.clone()).or_insert(0) += 1;
     }
+
+    block_info_counts
+        .into_iter()
+        .max_by_key(|(_block_info, count)| *count)
+        .map(|(block_info, _count)| block_info)
```

Comment:
:wrench: micro optimization, only clone once
```suggestion
    for block_info in block_infos {
        *block_info_counts.entry(block_info).or_insert(0) += 1;
    }

    block_info_counts
        .into_iter()
        .max_by_key(|(_block_info, count)| *count)
        .map(|(block_info, _count)| block_info.clone())
```